packages/vscode-extension/src/debugging/CDPSession.ts

```diff
@@ -148,6 +152,7 @@ export class CDPSession {
 
         if (isMainBundle) {
           this.debugSessionReady = true;
+          this.debugSessionReadyEmitter.fire({});
```

Comment:
given you added emitter maybe you can refactor `flishEnqueuedConsoleAPICalls` to just follow the same approach (instead of adding to queue we create event subscription for every log entry that needs to be scheduled)

Reply:
I'm not 100% sure the order of listeners is guaranteed, so I'd rather not.
We can still have a listener which flushes queue though, it should be cleaner.

Reply:
Extracted the "resolve a promise on event or reject after a timeout" part to a helper function, I think it reads better that way.

Reply:
Moved the body of this `if` statement to a listener on `onDebugSessionReady`